### PR TITLE
DDF-2116: NPM related build errors in Bamboo

### PR DESCRIPTION
**Pull request by lessarderic**
#### What does this PR do?
Update NPM and NodeJS versions to address build issues in Bamboo.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@coyotesqrl @ahoffer 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@pklinef
@shaundmorris
#### How should this be tested?
Build DDF
#### Any background context you want to provide?
The problem is caused by older versions of nodejs (<= 4.2.x) that fail on idle connections (see https://github.com/nodejs/node/issues/3595). Upgrading to a more recent version of nodejs that fixes issue #3595 addresses this issue.
#### What are the relevant tickets?
DDF-2116
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by pklinef** [comment]
:+1:

**Comment by shaundmorris** [comment]
@lessarderic can you summarize how the version upgrades resolve the build issues?

**Comment by lessarderic** [comment]
@shaundmorris, I just added a blurb in the PR description. Hopefully it addresses your comment. If not, let me know.

**Comment by shaundmorris** [comment]
thanks @lessarderic sounds good!

**Comment by kcwire** [comment]
when this works do we need to put up a PR for 2.9.x as well?

**Comment by lessarderic** [comment]
Bamboo build was successful but marked as failed because Bamboo ran out of memory parsing all of our surefire test result files (known issue).

@kcwire, I'll port this change to 2.9.x.

Merging.
